data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py

```diff
@@ -0,0 +1,261 @@
+                self.CSCORE_INPUT_FIELD: field_names.EPA_RSEI_CSCORE_OUTPUT_FIELD,
+                self.NCSCORE_INPUT_FIELD: field_names.EPA_RSEI_NCSCORE_OUTPUT_FIELD,
+                self.POPULATION_INPUT_FIELD: field_names.EPA_RSEI_POPULATION_OUTPUT_FIELD,
+            },
+            inplace=True,
+        )
+
+        # 5th, 25th, 50th, 75th and 99th percentiles for each score column
+        # this will yield n - 1 categorical variables where n = 5 percentiles specified
+        index = [5, 25, 50, 75, 99]
+        # Parameter for Cut function
+        labels_for_cut_function = [
+            "5 - 25 percentile values for score",
+            "25 - 50 percentile value for score",
+            "50 - 75 percentile value for score",
+            "75 - 99 percentile value for score",
+        ]
+
+        # Generate percentile values specified by index above
+        # Each is specific to the three different scores
+        perc_func_overall_risk_score = [
+            self._custom_percentile(
+                self.df[field_names.EPA_RSEI_SCORE_OUTPUT_FIELD].values, i
+            )
+            for i in index
+        ]
+
+        perc_func_cancer_score = [
+            self._custom_percentile(
+                self.df[field_names.EPA_RSEI_CSCORE_OUTPUT_FIELD].values, i
+            )
+            for i in index
+        ]
+
+        perc_func_non_cancer_score = [
+            self._custom_percentile(
+                self.df[field_names.EPA_RSEI_NCSCORE_OUTPUT_FIELD].values, i
+            )
+            for i in index
+        ]
+
+        # The rationale for creating these is located here https://www.epa.gov/rsei/understanding-rsei-results#what
+        # Section: "What does a high RSEI Score mean?"
+
+        # The cut function creates equispaced bins but frequency of samples is unequal in each bin
+        # It is certainly possible to apply qcut. qcut is a quantile based function to create bins
+        # Quantile is to divide the data into equal number of subgroups or probability distributions of
+        # equal probability into continuous interval. I opted for cut as it offers greater flexibility in
+        # creating percentile bins.
+        #
+        # The result produces a cateogrical variable that affords easier stratification
+        # when performing more advanced statistical analyses of each component of the distribution. This is pertienent
+        # if we seek to combine this data with population characteristics and demographics - for example, ACS and CDC SVI Index -
+        # in order to chracterize each section of the "higher scoring" tracts for cancer-related and non-cancer related toxicity
+        self.df[field_names.EPA_RSEI_SCORE_PERCENTILE_LABEL_FIELD] = pd.cut(
+            self.df[field_names.EPA_RSEI_SCORE_OUTPUT_FIELD],
+            right=True,  # include right-most interval
+            bins=perc_func_overall_risk_score,
+            labels=labels_for_cut_function,
+            ordered=True,
+        )
```

Comment:
I wanted to discuss this with @lucasmbrown-usds and discuss some of the initial thought processes above.

Reply:
Another way to stratify would be something like this:

```
def binning_function(col, cut_points, labels = None): 

    #Define min and max values:
    minval = col.min()
    maxval = col.max()

    # create list by adding min and max to cut_points
    break_points = [minval] + cut_points + [maxval]


    # if no labels provided, use
    # default labels 0...(n - 1)
    if not labels:
        labels = range(len(cut_points) + 1)

    # Binning using cut
    # function from Pandas
    column_binning = pd.cut(col, bins = break_points, labels = labels, include_lowest = True)
    return column_binning

# Binning overall risk score:
cut_points = [90, 140, 190]# hypothetical example of values
labels = ["low", "medium", "high", "very high"]
self.df[field_names.EPA_RSEI_SCORE_LABEL_FIELD] = binning_function(self.df[field_names.EPA_RSEI_SCORE_OUPUT_FIELD], cut_points, labels)
```